azure-functions-gradle-plugin/src/main/java/com/microsoft/azure/plugin/functions/gradle/AzureFunctionsPlugin.java

```diff
@@ -81,11 +76,25 @@ public void apply(final Project project) {
         });
 
         project.afterEvaluate(projectAfterEvaluation -> {
-
             mergeCommandLineParameters(extension);
-            TelemetryAgent.getInstance().initTelemetry(GRADLE_PLUGIN_NAME,
-                StringUtils.firstNonBlank(AzureFunctionsPlugin.class.getPackage().getImplementationVersion(), "develop"), // default version: develop
-                BooleanUtils.isNotFalse(extension.getAllowTelemetry()));
+
+            AzureMessager.setDefaultMessager(new GradleAzureMessager(project.getLogger()));
+            Azure.az().config().setLogLevel(HttpLogDetailLevel.NONE.name());
+            Azure.az().config().setProduct(GRADLE_PLUGIN_NAME);
+            final String pluginVersion = StringUtils.firstNonBlank(AzureFunctionsPlugin.class.getPackage().getImplementationVersion(), "develop");
+            Azure.az().config().setVersion(pluginVersion);
+            final boolean enableTelemetry = BooleanUtils.isNotFalse(extension.getAllowTelemetry());
+            Azure.az().config().setTelemetryEnabled(enableTelemetry);
+            final String installationId = InstallationIdUtils.getHashMac();
+            Azure.az().config().setMachineId(installationId);
+            final String sessionId = UUID.randomUUID().toString();
+            Azure.az().config().setSessionId(sessionId);
+            final String userAgent = enableTelemetry ?
+                    String.format("%s/%s %s:%s %s:%s", GRADLE_PLUGIN_NAME, pluginVersion,
+                            INSTALLATION_ID_KEY, installationId, SESSION_ID_KEY, sessionId)
+                    : String.format("%s/%s", GRADLE_PLUGIN_NAME, pluginVersion);
+            Azure.az().config().setUserAgent(userAgent);
+
             TelemetryAgent.getInstance().showPrivacyStatement();
```

Comment:
The above looks all good. The question I have is if you really want to show the TelemetryAgent's privacy statement if telemetry is disabled. I've modified my build to write the ~/.azure/gradlePlugin.properties and added the first.run=false.

Reply:
@enaess Thank you for your comment and sorry for the delayed reply. The purpose of this logic is to display the statement only once when users run the gradle toolkits for the first time. Otherwise, the statement would appear every time they run the toolkit, which might be annoying.